pyvista/core/filters.py

```diff
@@ -663,7 +663,7 @@ def elevation(dataset, low_point=None, high_point=None, scalar_range=None,
 
     def contour(dataset, isosurfaces=10, scalars=None, compute_normals=False,
                 compute_gradients=False, compute_scalars=True, rng=None,
-                preference='point'):
+                preference='point', method=None):
```

Comment:
Instead of defaulting to `None`, let's just default to `'contour'`

Reply:
Or does another one of these algorithms perform better that we should default to?

Reply:
the standard contour default is probably safest though

Reply:
For my purposes I didn't see any significant difference between them.

I'll switch to `contour` then. I generally like `None` for default arguments, because it allows easy propagation of the default value when extending (i.e. creating anything that wraps this method and also wants to expose `method`) or scripting. E.g. if I want to use the default value unless some condition is met I can do `method=None; if special_case: method=marching_cubes; obj.contour(method=method)`.

Reply:
That's a fair point!

Throughout PyVista's API, we have tons of default arguments set to `None` but they typically lookup a choice/values from somewhere else or a preference from `rcParams` if `None`. I'd like to keep it that way for consistency, but you could still leave the `if method is None or method == "contour":` statement to catch that case.